xds/internal/resolver/watch_service.go

```diff
@@ -84,16 +96,30 @@ func (w *serviceUpdateWatcher) handleLDSResp(update xdsclient.ListenerUpdate, er
 			w.rdsCancel()
 			w.rdsName = ""
 			w.rdsCancel = nil
+			w.lastUpdate = serviceUpdate{}
 		}
 		// The other error cases still return early without canceling the
 		// existing RDS watch.
 		w.serviceCb(serviceUpdate{}, err)
 		return
 	}
 
+	sendUpdate := false
+	ldsUpdate := ldsConfig{maxStreamDuration: update.MaxStreamDuration}
+	if w.lastUpdate.ldsConfig != ldsUpdate {
+		w.lastUpdate.ldsConfig = ldsUpdate
+		sendUpdate = true
```

Comment:
We also shouldn't send the update if this LDS update contains a new RDS name to watch.
Move this if check inside `if w.rdsName == update.RouteConfigName`?

Reply:
Refactored a bit. PTAL